examples/reflection/trait_reflection.rs

```diff
@@ -45,7 +45,7 @@ fn setup(type_registry: Res<TypeRegistry>) {
     // The #[reflect] attribute we put on our DoThing trait generated a new `ReflectDoThing` struct,
     // which implements TypeData. This was added to MyType's TypeRegistration.
     let reflect_do_thing = type_registry
-        .get_type_data::<ReflectDoThing>(reflect_value.type_id())
+        .get_type_data::<ReflectDoThing>(Reflect::type_id(&*reflect_value))
```

Comment:
Is `reflect_value.type_id()` causing an ambiguity error? Are we forced to use `Reflect::type_id(&*reflect_value)`?

Reply:
Here, yes, it is... for some reason. I don't think std::any::Any is really in scope but for some reason it was ambiguous. It doesn't always happen, but if Any is in scope then it does. I think it's much clearer this way anyway though.